pool/src/transaction_pool.rs

```diff
@@ -192,18 +192,15 @@ impl TransactionPool {
 		let bucket_transactions = self.txpool.bucket_transactions(Weighting::NoLimit);
 
 		// Get last transaction and remove it
-		match bucket_transactions.last() {
-			Some(evictable_transaction) => {
-				// Remove transaction
-				self.txpool.entries = self
-					.txpool
-					.entries
-					.iter()
-					.filter(|x| x.tx != *evictable_transaction)
-					.map(|x| x.clone())
-					.collect::<Vec<_>>();
-			}
-			None => (),
+		if let Some(evictable_transaction) = bucket_transactions.last() {
+			// Remove transaction
+			self.txpool.entries = self
```

Comment:
It would be more efficient to do this in-place using [`retain`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.retain), instead of cloning each element and then replacing the original vector

Reply:
Nice! Everyday a new function in Rust world :)